packages/react/src/utils/hashKey.ts

```diff
@@ -0,0 +1,44 @@
+import { Key } from '../experimental/Await'
+
+export const hashKey = (key: Key) => {
+  return JSON.stringify(key, (_, val) =>
+    isPlainObject(val)
+      ? Object.keys(val)
+          .sort()
+          .reduce((result, key) => {
+            result[key] = val[key]
+            return result
+          }, {} as any)
+      : val
+  )
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const isPlainObject = (o: any): o is Object => {
```

Comment:
about **hashing**, wouldn't it be more rigorous to check with `type PlainObject = Record<string, any>` instead of just an object type?

```suggestion
type PlainObject = Record<string, any>

export const hashKey = (key: Key) => {
  return JSON.stringify(key, (_, val) =>
    isPlainObject(val)
      ? Object.keys(val)
          .sort()
          .reduce((result: PlainObject, key) => {
            result[key] = val[key]
            return result
          }, {})
      : val
  )
}

const isPlainObject = (o: any): o is PlainObject => {
```

we can also eliminate the lint directive `// eslint-disable-next-line @typescript-eslint/ban-types`.